api/v1beta1/awsloadbalancercontroller_types.go

```diff
@@ -0,0 +1,248 @@
+	// requested using the Cloud Credentials API,
+	// see https://docs.openshift.com/container-platform/4.11/authentication/managing_cloud_provider_credentials/about-cloud-credential-operator.html.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Credentials *SecretReference `json:"credentials,omitempty"`
+}
+
+// AWSResourceTag is a tag to apply to AWS resources created by the controller.
+type AWSResourceTag struct {
+	// key is the key of the tag.
+	// See https://docs.aws.amazon.com/tag-editor/latest/userguide/tagging.html#tag-conventions
+	// for information on the tagging conventions.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=128
+	// +kubebuilder:validation:Pattern=`^[0-9A-Za-z_.:/=+-@]+$`
+	// +required
+	Key string `json:"key"`
+
+	// value is the value of the tag.
+	// See https://docs.aws.amazon.com/tag-editor/latest/userguide/tagging.html#tag-conventions
+	// for information on the tagging conventions.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:Pattern=`^[0-9A-Za-z_.:/=+-@]*$`
+	// +required
+	Value string `json:"value"`
+}
+
+type AWSLoadBalancerDeploymentConfig struct {
+	// +kubebuilder:default:=2
+	// +kubebuilder:validation:Optional
+	// +optional
+	Replicas int32 `json:"replicas,omitempty"`
+}
+
+// SecretReference contains the information to let you locate the desired secret.
+// Secret is required to be in the operator namespace.
+type SecretReference struct {
+	// name is the name of the secret.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+}
+
+// AWSLoadBalancerControllerStatus defines the observed state of AWSLoadBalancerController.
+type AWSLoadBalancerControllerStatus struct {
+	// conditions is a list of operator-specific conditions
+	// and their status.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	// observedGeneration is the most recent generation observed.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// subnets contains details of the subnets of the cluster
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Subnets *AWSLoadBalancerControllerStatusSubnets `json:"subnets,omitempty"`
+
+	// ingressClass is the current default Ingress class.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	IngressClass string `json:"ingressClass,omitempty"`
+}
+
+type AWSLoadBalancerControllerStatusSubnets struct {
+	// subnetTagging indicates the current status of the subnet tags
+	// +kubebuilder:validation:Optional
+	// +optional
+	SubnetTagging SubnetTaggingPolicy `json:"subnetTagging,omitempty"`
```

Comment:
Is `SubnetTagging` field in the status necessary given there's nothing to reconcile in the cluster wrt. the tagging policy? When `.spec.subnetTagging` is set, the operator starts operating/tagging over the subnets. What additional value is `.spec.subnetTagging` adding? Is `.status.subnetTagging` changed right away after `.spec.subnetTagging` is? Or, is there some additional work/activity performed right after `.spec.subnetTagging` changes but before `.status.subnetTagging` gets updated?

Reply:
> Is .status.subnetTagging changed right away after .spec.subnetTagging is?

No.

> Or, is there some additional work/activity performed right after .spec.subnetTagging changes but before .status.subnetTagging gets updated?

Yes, the subnet tagging may happen and if it succeeds the status gets set to the same value as in `.spec.subnetTagging`.

> Is SubnetTagging field in the status necessary given there's nothing to reconcile in the cluster wrt. the tagging policy?

Well, it's there to let the user know that their request of the policy changing has been successfully applied.
 
> When .spec.subnetTagging is set, the operator starts operating/tagging over the subnets. What additional value is .spec.subnetTagging adding?

I would say that's the whole value - make sure the required subnets are properly tagged if in `Auto` and let the user do the tagging by themselves if in `Manual`.